app/src/main/java/com/example/calculator/Calculation.java

```diff
@@ -5,13 +5,57 @@
 public class Calculation {
 
     private ArrayList<Token> mStringArrayList;
+    private double answer;
+    private boolean errorOccurred;
+    private String errorMessage;
 
     public Calculation(ArrayList<Character> arr) {
-         mStringArrayList = setTokenArrayList(arr);
+        errorOccurred = false;
+        errorMessage = "No Error";
+        mStringArrayList = setTokenArrayList(arr);
+        pairParenthesis();
+        answer = solveAnswer();
     }
 
-    public double solveAnswer() {
+    public double getAnswer() {
+        return answer;
+    }
+
+    public boolean isErrorOccurred() {
+        return errorOccurred;
+    }
+
+    public String getErrorMessage() {
```

Comment:
ErrorMesage itself is a representation.
So, the class should return some error codes, and the call should decide the message.